File: go/prob2.go
package main

import "fmt"

func main() {
	var result, first, second, sum int = 0, 0, 1, 0
	for ; result < 4000000; {
		result = first + second
		first = second
		second = result
		if(result % 2 == 0){
			sum += result
		}
	}
    fmt.Printf("Sum: %v\n", sum)
}

File: java/prob2.java
public class prob2
{
	public static void main(String[] args)
	{
		int sum = 0, result = 0, first = 0, second = 1;
		while(result <= 4000000)
		{
			result = first + second;
			first = second;
			second = result;
			if( result % 2 == 0)
			{
				sum += result;
			}
		}
		System.out.println("Sum: "+sum);
	}
}